packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -600,7 +609,7 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
     private getOutboundRoutes(): string[] {
         const outboundRoutes: string[] = [];
         for (const [contextId] of this.contexts) {
-            outboundRoutes.push(`${this.absolutePath}/${contextId}`);
```

Comment:
@agarwal-navin , can you please take a look at changes in this file? Related to routing & GC.
Basically 2 places where this.absolutePath was used before

Reply:
If I understand things correctly, the absolute paths for handles will now look like this:
- For a data store (say datastore1) - `/_channels/datastore1`.
- For a DDS (say dds1) in datastore1 - `/_channels/datastore1/_channels/dds1`.

So, in this piece of code, the route added to outboundRoutes should be - `/_channels/<datastoreId>/_channels/<contextId>`

What will `this.channelsRoutingContext.absolutePath` return?

Reply:
In addition to this, we need to do the following in `getGCData`:
1. DataStoreRuntime needs to prefix `_channels` to all routes returned by the child channel contexts.
2. ContainerRuntime needs to prefix `_channels` to all routes returned by the child data stores.

This can be done by passing the `_channels/<childrenId>` prefix when calling `prefixAndAddNodes`.

Reply:
Data store has no handle representation, so /_channels/<datastoreid> should never show up, with exception of GC (where data store is a node from GC point of view). Otherwise yes, the routes are correct.
Nobody should ever construct URIs manually (i.e. prefix with _channels). We should use routing context path always, never add or remove to / from it. It represents the truth, via organizing a chain of contexts that form a path ( sort of violate my own rule here by adding contextId, but knowledge how paths for channels are formed is in the same layer / file, so that probably Ok).

this.channelsRoutingContext.absolutePath will return a path used to represent channels. As a general rule - data store does not know what this path is, as it might be hosted by who knows whom (in future - by other data stores). But from practical perspective today that would result in /_channels/<datastoreId>/_channels/ path.

I believe this code is correct, but you bring a good point - all places where prefixAndAddNodes needs to be fixed (appending /_channels). This seems unfortunate as how paths are formed has to be encoded in two places - something that I really wanted to avoid (and removed in other places, leaving only one source of truth). I'd prefer to move towards absolute paths across the board. I'll make that fix, but let's keep chatting about long term design here.